tests/sentry/api/endpoints/test_doc_integration_details.py

```diff
@@ -199,15 +199,46 @@ def test_update_empty_metadata(self):
 
     def test_update_ignore_keys(self):
         """
-        Test that certain reserved keys cannot be overridden by the
+        Tests that certain reserved keys cannot be overridden by the
         request payload. They must be created by the API.
         """
         self.login_as(user=self.superuser, superuser=True)
         payload = {**self.payload, "metadata": {"should": "not override"}}
-        self.get_success_response(self.doc_1.slug, status_code=status.HTTP_200_OK, **payload)
+        self.get_success_response(self.doc_2.slug, status_code=status.HTTP_200_OK, **payload)
         # Ensure the DocIntegration was not created with the ignored keys' values
         for key in self.ignored_keys:
-            assert getattr(self.doc_1, key) is not payload[key]
+            assert getattr(self.doc_2, key) is not payload[key]
+
+    def test_update_simple_without_avatar(self):
+        """
+        Tests that the DocIntegration can be edited without an
+        associated DocIntegrationAvatar.
+        """
+        self.login_as(user=self.superuser, superuser=True)
+        payload = {**self.payload, "is_draft": True}
+        response = self.get_success_response(
+            self.doc_1.slug, status_code=status.HTTP_200_OK, **payload
+        )
+        self.doc_1.refresh_from_db()
+        assert serialize(self.doc_1) == response.data
+
+    def test_update_publish_without_avatar(self):
+        """
+        Tests that the DocIntegration cannot be published without an
+        associated DocIntegrationAvatar.
+        """
+        self.login_as(user=self.superuser, superuser=True)
+        response = self.get_error_response(
+            self.doc_1.slug, status_code=status.HTTP_400_BAD_REQUEST, **self.payload
+        )
+        assert "avatar" in response.data.keys()
```

Comment:
What's this check for?

Reply:
This checks that there was an error on the 'avatar' field. It's just the format that validation errors send back from the serializer.